webpack.config.ts

```diff
@@ -0,0 +1,234 @@
+							plugins: [require('autoprefixer')],
+						},
+					},
+					'sass-loader',
+				],
+			},
+			{
+				test: /\.css$/,
+				use: [
+					{
+						loader: MiniCssExtractPlugin.loader,
+						options: {
+							hmr: !prod,
+							sourceMap: !prod || sourceMapsInProduction,
+						},
+					},
+					'css-loader',
+				],
+			},
+			{
+				test: /\.ts$/,
+				use: 'ts-loader',
+				exclude: /node_modules/,
+			},
+		],
+	},
+	devServer: {
+		hot: true,
+		stats: 'minimal',
+		contentBase: 'public',
+		watchContentBase: true,
+	},
+	mode,
+	plugins: [
+		new MiniCssExtractPlugin({
+			filename: '[name].css',
+		}),
+	],
+	optimization: {
+		minimizer: [],
+	},
+	devtool: prod && !sourceMapsInProduction ? false : 'source-map',
+};
+
+// Add stylesheets to the build
+if (Array.isArray(stylesheets) || typeof stylesheets === 'string') {
+	if (!Array.isArray(stylesheets)) {
+		stylesheets = [stylesheets];
+	}
+
+	// Explicitly type bundle as string[]
+	type Replace<T, K> = Omit<T, keyof K> & K;
+	config.entry = config.entry as Replace<
+		typeof config.entry,
+		{
+			bundle: string[];
+		}
+	>;
+	(config.entry.bundle as string[]).unshift.apply(
+		config.entry.bundle,
+		stylesheets
+	);
+}
+
+// Load path mapping from tsconfig
+const tsconfigPath = path.resolve(__dirname, 'tsconfig.json');
+const tsconfig = require('fs').existsSync(tsconfigPath)
+	? require(tsconfigPath)
+	: {};
+if ('compilerOptions' in tsconfig && 'paths' in tsconfig.compilerOptions) {
+	const aliases = tsconfig.compilerOptions.paths;
+	for (const alias in aliases) {
+		const paths = aliases[alias].map((p) => path.resolve(__dirname, p));
+
+		if (!(alias in config.resolve.alias) && paths.length) {
+			config.resolve.alias[alias] = paths.length > 1 ? paths : paths[0];
+		}
+	}
+}
+
+// These options should only apply to production builds
+if (prod) {
+	// Clean the build directory for production builds
+	config.plugins.push(new CleanWebpackPlugin());
+
+	// Minify CSS
+	config.optimization.minimizer.push(
+		new OptimizeCSSAssetsPlugin({
+			cssProcessorOptions: {
+				map: sourceMapsInProduction
+					? {
+							inline: false,
+							annotation: true,
+						}
+					: false,
+			},
+			cssProcessorPluginOptions: {
+				preset: [
+					'default',
+					{
+						discardComments: {
+							removeAll: !sourceMapsInProduction,
+						},
+					},
+				],
+			},
+		})
+	);
+
+	// Minify and treeshake JS
+	config.optimization.minimizer.push(
+		new TerserPlugin({
+			sourceMap: sourceMapsInProduction,
+			extractComments: false,
+		})
+	);
+}
+
+// Add babel if enabled
+if (useBabel && (prod || useBabelInDevelopment)) {
+	config.module.rules.unshift({
+		test: /\.(?:svelte|m?js)$/,
+		include: [path.resolve(__dirname, 'src'), path.dirname(sveltePath)],
+		use: {
+			loader: 'babel-loader',
+			options: {
+				sourceType: 'unambiguous',
+				presets: ['@babel/preset-env'],
+				plugins: ['@babel/plugin-transform-runtime'],
+			},
+		},
+	});
+}
+
+export default config;
```

Comment:
I tried the align the whitespace with the project conventions (I use spaces but the project uses tabs) but I failed :O If you want to move forward with the PR maybe you can pull the code and format it with your linter so that the whitespace is in line with the project?